### PR TITLE
fix: display organizer location after booking

### LINKED ISSUE #11455: [CAL-2497] Organizer address is not showing in the confirmation email
* Set up organizer address as the location
* Book a meeting
* it just says "In person (Organizer Address) without the listed address in the location field

Created via Threads. See full discussion: [https://threads.com/34537580183](https://threads.com/34537580183)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2497](https://linear.app/calcom/issue/CAL-2497/organizer-address-is-not-showing-in-the-confirmation-email)</sub>

<sub>[CAL-2498](https://linear.app/calcom/issue/CAL-2498/[cal-2497]-organizer-address-is-not-showing-in-the-confirmation-email)</sub>

### LINKED ISSUE #12097: [CAL-2651] Adding Organizer input type Locations in event setup page
Except In Person Address. Other Organizer input type locations have this issue

https://github.com/calcom/cal.com/assets/53316345/77dfb0bb-9ec4-4188-8e76-85c54ac917a2

<sub>[CAL-2651](https://linear.app/calcom/issue/CAL-2651/adding-organizer-input-type-locations-in-event-setup-page)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
Fixes: https://github.com/calcom/cal.com/issues/11455
Fixes: https://github.com/calcom/cal.com/issues/12097

Added E2E Test

Screenshot 2023-10-26 at 1 30 33 AM

How to test?

1) Create an event type with location In Person (Organzier Address) and add address and click save
2) Make sure display booking option is false.
3) Go to booking page
4) Make sure address is not visible on booking page.
5) Book any time slot
6) Make sure now address is visible on the success page and email

### CODE REVIEW

File: packages/features/form-builder/FormBuilderField.tsx [hunk 1]
```diff
@@ -311,8 +312,20 @@ export const ComponentForField = ({
...
+
     const options = field.options.map((field) => {
-      return { ...field, value: field.value === "inPerson" ? field.label : field.value };
+      return {
+        ...field,
+        value:
+          field.value === DefaultEventLocationTypeEnum.InPerson && moreThanOneInPersonOptions
+            ? field.label
+            : field.value,
+      };
```
Comment (hariombalhara) [inline packages/features/form-builder/FormBuilderField.tsx:328]: FormBuilderField shouldn't know about Locations, it's a generic component to show any kind of FormBuilder field

**Review by hariombalhara — CHANGES_REQUESTED** [review]
@Udit-takkar Left a blocking comment, we need a better way to solve this.

File: apps/web/components/eventtype/EventSetupTab.tsx [hunk 2]
```diff
@@ -256,21 +256,22 @@ export const EventSetupTab = (
...
+              if (eventLocationType?.organizerInputType) {
+                return (
+                  location.type === eventLocationType?.type &&
+                  location[eventLocationType.defaultValueVariable] ===
+                    field?.[eventLocationType.defaultValueVariable]
+                );
+              } else {
+                return location.type === eventLocationType?.type;
+              }
+            });
```
Comment (hariombalhara) [inline apps/web/components/eventtype/EventSetupTab.tsx:274]: 1. This seems like abstractable to a function. 
2. I am having a hard time understanding why we are using both `locationFields` and `formMethod.getValues('locations')`, they seem to be having the same value.
3. What is the case when the two would be different?
4. What is the case when `location.type === eventLocationType?.type` will be false.

File: packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx [hunk 3]
```diff
@@ -105,6 +106,24 @@ export const BookingFields = ({
...
+            field.options.filter((field) => {
+              return field.value === DefaultEventLocationTypeEnum.InPerson;
+            }).length > 1
+          );
+
+          field.options = field.options.map((field) => {
+            return {
+              ...field,
+              value:
+                field.value === DefaultEventLocationTypeEnum.InPerson && moreThanOneInPersonOptions
```
Comment (hariombalhara) [inline packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:120]: I think I saw similar logic elsewhere but trying to understand.
1. Why can't we use the` field.label ` approach in all cases? 
2. Wouldn't the same applicable to other `organizerInput` type locations and not just InPerson

Comment (Udit-takkar) [reply packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:120]: yes we should check for other organizer input type also. forgot to update this

File: packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx [hunk 4]
```diff
@@ -105,6 +106,29 @@ export const BookingFields = ({
           );
         }
 
+        if (field?.options) {
+          const organzierInputTypes = getOrganizerInputLocationsType();
```
Comment (hariombalhara) [inline packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:110]: typo: `organizerInputTypes`

File: packages/app-store/locations.ts [hunk 5]
```diff
@@ -427,3 +427,11 @@ export const getTranslatedLocation = (
 
   return translatedLocation;
 };
+
+export const getOrganizerInputLocationsType = () => {
+  const locationTypes: DefaultEventLocationType["type"] | EventLocationTypeFromApp["type"][] = [];
+  const locations = locationsTypes.filter((location) => !!location.organizerInputType);
+  locations?.forEach((l) => locationTypes.push(l.type));
```
Comment (hariombalhara) [inline packages/app-store/locations.ts:434]: Naming variables in locations related code is a Pain.  I am seeing `locationsTypes` and `locationTypes` 😂

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**